### PR TITLE
#595 Add generator transformer model

### PR DESCRIPTION
**Pull request by MarianneSaugierRTE**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo [hunk 1]
```diff
@@ -160,4 +160,61 @@ equation
...
+    Types.Angle U2Phase0 "Start value of voltage angle in rad";
+
+  equation
+
+    s10Pu = Complex(P10Pu, Q10Pu);
+    u10Pu = ComplexMath.fromPolar(U10Pu, U1Phase0);
+    s10Pu = u10Pu * ComplexMath.conj(i10Pu);
+
+    rTfoPu * rTfoPu * u10Pu = rTfoPu * u20Pu + ZPu * i10Pu;
+    i10Pu = rTfoPu * (YPu * u20Pu - i20Pu);
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: Again here the physical part (the two equations above) are the same in all the transformer initialization mode. Should we factorize them somewhere or keep them duplicated in different base models?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/GeneratorTransformer.mo [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  parameter Types.ComplexVoltagePu u10Pu "Start value of complex voltage at terminal 1 (base U1Nom)";
+  parameter Types.ComplexCurrentPu i10Pu "Start value of complex current at terminal 1 (base U1Nom, SnRef) (receptor convention)";
+  parameter Types.ComplexVoltagePu u20Pu "Start value of complex voltage at terminal 2 (base U2Nom)";
+  parameter Types.ComplexCurrentPu i20Pu "Start value of complex current at terminal 2 (base U2Nom, SnRef) (receptor convention)";
+
+equation
+
+  if (running.value) then
+    rTfoPu * rTfoPu * terminal1.V = rTfoPu * terminal2.V + ZPu * terminal1.i;
+    terminal1.i = rTfoPu * (YPu * terminal2.V - terminal2.i);
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/GeneratorTransformer.mo]: Same remark as for the initialization model

File: dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp [hunk 3]
```diff
@@ -57,8 +57,8 @@ GeneratorInterfaceIIDM::getComponentVarIndex(const std::string& varName) const {
 void
 GeneratorInterfaceIIDM::exportStateVariablesUnitComponent() {
   bool connected = (getValue<int>(VAR_STATE) == CLOSED);
-  generatorIIDM_.p(-1 * getValue<double>(VAR_P) * SNREF);
-  generatorIIDM_.q(-1 * getValue<double>(VAR_Q) * SNREF);
+  generatorIIDM_.p(getValue<double>(VAR_P) * SNREF);
+  generatorIIDM_.q(getValue<double>(VAR_Q) * SNREF);
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp]: yes

[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp]: This modification is not compatible with the convention of the cpp generator model, I revert it

File: dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp [hunk 4]
```diff
@@ -57,8 +57,8 @@ GeneratorInterfaceIIDM::getComponentVarIndex(const std::string& varName) const {
...
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Modeler/DataInterface/DYNGeneratorInterfaceIIDM.cpp]: revert

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/GeneratorSynchronousExtTfo_3E_INIT.mo [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model GeneratorSynchronousExtTfo_3E_INIT "Synchronous machine with 3 windings - Initialization model from external parameters"
+
+/*This model is similar to GeneratorSynchronousExt_3E_INIT but U0Pu, P0Pu, Q0Pu and UPhase0 are variables because they are calculated from the generator transformer initialization model and should be passed to the generator through an initConnect in a preassembled model or in the dyd file*/
+
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/GeneratorSynchronousExtTfo_3E_INIT.mo]: it is good to mutualize when possible, therefore I created new base classes in BaseClasses_INIT

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo [hunk 6]
```diff
@@ -160,4 +160,61 @@ equation
...
+partial model BaseGeneratorTransformer_INIT "Base model for initialization of GeneratorTransformer"
+
+/*
+  This model enables to initialize the generator model when the load-flow inputs are not known at the generator terminal but at the generator transformer terminal.
+
+  Equivalent circuit and conventions:
+
+               I1  r                I2
+    U1,P1,Q1 -->---oo----R+jX-------<-- U2,P2,Q2
+  (terminal1)                   |      (terminal2)
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: done

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo (hunk 1 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: factorization done

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/GeneratorTransformer.mo (hunk 2 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/GeneratorTransformer.mo]: factorization done

**Pull request merged**